### PR TITLE
FEAT: Add Watchlater management

### PR DESCRIPTION
**Pull request by VishalPatil18**
### 📝 Description

Watch later is a list of videos that the user can save for watching videos later or for future reference.

### 🚀 New behavior

The user can now
- Add Videos to watchlater
- Remove them from watchlater
- Remove all videos from watchlater
- Remove all videos from history
- Remove all videos from likes

### 🖼 Preview Link

https://vistream-react-version-6w73cupan-vishalpatil18.vercel.app/

### 💣 Is this a breaking change?
- [ ] Yes
- [x] No

### 🔬 Is the code self-reviewed?
- [x] Yes
- [ ] No

### 📋 Is the code well-formatted?
- [x] Yes
- [ ] No

### 🔗 Related to Issue 
- #25 

> NOTE: Trending videos is not yet implemented on the home page

### CODE REVIEW

File: src/components/SmallVideoCard/SmallVideoCard.jsx [hunk 1]
```diff
@@ -98,7 +119,9 @@ const SmallVideoCard = ({ video, page = "", playlistID = "" }) => {
                 })()}
                 className={styles.menuActionBtnIcon}
               />
-              {(page === "liked" && isVideoLiked) || true
+              {(page === "liked" && isVideoLiked) ||
+              (page === "watchlater" && inWatchlater) ||
+              true
                 ? `Remove from ${page}`
                 : `Add to ${page}`}
```
Comment (AtulPant2704) [inline src/components/SmallVideoCard/SmallVideoCard.jsx:126]: Really nice use of ternary.

File: src/utilities/auth/loginHandler.js [hunk 2]
```diff
@@ -68,6 +70,15 @@ const loginHandler = async (
...
+      const responseWatchlater = await getWatchlaterService(
+        response.data.encodedToken
+      );
+      watchlaterDispatch({
+        type: "INITIAL_WATCHLATER",
+        payload: {
+          watchlater: responseWatchlater.data.watchlater,
+        },
+      });
       componentsDispatch({
```
Comment (AtulPant2704) [inline src/utilities/auth/loginHandler.js:82]: Check here for response. status and then send a dispatch.

**Review by AtulPant2704 — APPROVED** [review]
Functionality is working fine and code is very well modularised, good job.

**Review by kart-c — APPROVED** [review]
Functionality is working fine @VishalPatil18 
- I like that you have extracted all the repeating functions to make it easy to review
- Very clean code

**Comment by AnkurChunekar** [comment]
vishal pr

- On your Homepage the video menu should display `Add to watchlater` if video isn't added already, instead it displays REMOVE from watch later for both cases

- Also i received a 409 error alert, when I tried to remove a video from watchlater from homepage.

File: src/utilities/auth/loginHandler.js (hunk 2 above)
Comment (VishalPatil18) [reply src/utilities/auth/loginHandler.js:82]: Thanks for pointing it out! I might have missed that. I'll correct that.

**Comment by VishalPatil18** [comment]
> vishal pr
> 
>     * On your Homepage the video menu should display `Add to watchlater` if video isn't added already, instead it displays REMOVE from watch later for both cases
> 
>     * Also i received a 409 error alert, when I tried to remove a video from watchlater from homepage.

The trending Videos section has not been implemented yet! I had mentioned it in the PR description also as a note 👀

**Pull request merged**